python/cudf/cudf/_lib/aggregation.pyx

```diff
@@ -844,7 +565,202 @@ cdef class GroupbyScanAggregation:
+        cdef ReduceAggregation agg = cls()
+
+        if not pd.api.types.is_list_like(q):
+            q = [q]
+
+        cdef vector[double] c_q = q
+        cdef libcudf_types.interpolation c_interp = (
+            <libcudf_types.interpolation> (
+                <underlying_type_t_interpolation> (
+                    Interpolation[interpolation.upper()]
+                )
+            )
+        )
+        agg.c_obj = move(
+            libcudf_aggregation.make_quantile_aggregation[reduce_aggregation](
+                c_q, c_interp)
+        )
+        return agg
+
+    @classmethod
+    def nunique(cls):
+        cdef ReduceAggregation agg = cls()
+        agg.c_obj = move(
+            libcudf_aggregation.make_nunique_aggregation[reduce_aggregation]())
+        return agg
+
+    @classmethod
+    def nth(cls, libcudf_types.size_type size):
+        cdef ReduceAggregation agg = cls()
+        agg.c_obj = move(
+            libcudf_aggregation.make_nth_element_aggregation[
+                reduce_aggregation](size))
+        return agg
+
+cdef class ScanAggregation:
+    """A Cython wrapper for scan aggregations.
+
+    **This class should never be instantiated using a standard constructor,
+    only using one of its many factories.** These factories handle mapping
+    different cudf operations to their libcudf analogs, e.g.
+    `cudf.DataFrame.idxmin` -> `libcudf.argmin`. Additionally, they perform
+    any additional configuration needed to translate Python arguments into
+    their corresponding C++ types (for instance, C++ enumerations used for
+    flag arguments). The factory approach is necessary to support operations
+    like `df.agg(lambda x: x.sum())`; such functions are called with this
+    class as an argument to generation the desired aggregation.
+    """
+    @property
+    def kind(self):
+        return AggregationKind(self.c_obj.get()[0].kind).name
+
+    @classmethod
+    def sum(cls):
+        cdef ScanAggregation agg = cls()
+        agg.c_obj = move(
+            libcudf_aggregation.
+            make_sum_aggregation[scan_aggregation]())
+        return agg
+
+    @classmethod
+    def product(cls):
+        cdef ScanAggregation agg = cls()
+        agg.c_obj = move(
+            libcudf_aggregation.make_product_aggregation[scan_aggregation]())
+        return agg
+    prod = product
+
+    @classmethod
+    def min(cls):
+        cdef ScanAggregation agg = cls()
+        agg.c_obj = move(
+            libcudf_aggregation.
+            make_min_aggregation[scan_aggregation]())
+        return agg
+
+    @classmethod
+    def max(cls):
+        cdef ScanAggregation agg = cls()
+        agg.c_obj = move(
+            libcudf_aggregation.
+            make_max_aggregation[scan_aggregation]())
+        return agg
+
+    # scan aggregations
+    # TODO: update this after adding per algorithm aggregation derived types
+    # https://github.com/rapidsai/cudf/issues/7106
```

Comment:
The linked issue #7106 has been resolved -- is this TODO still relevant? Same in `GroupbyScanAggregation` above.

Reply:
Linked issue is for C++/libcudf only. Cython still need to update. This PR is actually the required update.